### PR TITLE
GH-2442 value comparator benchmarks and equals

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #2442  

Briefly describe the changes proposed in this PR:

 - benchmarks
 - removed .equals use from the ValueComparator

---- 
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [ ] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by hmottestad** [comment]
# Benchmark results

## Before
```
Benchmark                                       Mode  Cnt    Score    Error  Units
ValueComparatorBenchmark.sortManyDeepEquals     avgt    5  452.960 ± 29.174  ms/op
ValueComparatorBenchmark.sortManyPointerEquals  avgt    5  157.041 ±  6.604  ms/op
ValueComparatorBenchmark.sortObjects            avgt    5  250.146 ±  3.246  ms/op
ValueComparatorBenchmark.sortPredicates         avgt    5    0.107 ±  0.011  ms/op
ValueComparatorBenchmark.sortSubjects           avgt    5  104.293 ±  0.666  ms/op
```

## After
```
Benchmark                                       Mode  Cnt    Score    Error  Units
ValueComparatorBenchmark.sortManyDeepEquals     avgt   10  485.574 ± 18.843  ms/op
ValueComparatorBenchmark.sortManyPointerEquals  avgt   10  132.606 ±  0.654  ms/op
ValueComparatorBenchmark.sortObjects            avgt   10  207.202 ±  3.310  ms/op
ValueComparatorBenchmark.sortPredicates         avgt   10    0.093 ±  0.001  ms/op
ValueComparatorBenchmark.sortSubjects           avgt   10   79.736 ±  0.494  ms/op
```

**Comment by hmottestad** [comment]
The goal here was to simplify the comparison code. Essentially, the complexity for .equals(...) is similar to doing a comparison. This means that we can remove the .equals and just rely on the comparison code. The .equals(...) is inside the call to ObjectUtil.nullEquals(...). We still check for simple pointer equality.

**Comment by hmottestad** [comment]
Seems to give a decent ~20% performance improvement without any significant performance hits.

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/util/benchmark/ValueComparatorBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
+/**
+ * ****************************************************************************
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/util/benchmark/ValueComparatorBenchmark.java:2]: The license header looks like it got mangled a bit

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/util/benchmark/ValueComparatorBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+@Measurement(iterations = 10)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ValueComparatorBenchmark {
+
+	List<Value> subjects;
+	List<Value> predicates;
+	List<Value> objects;
+	List<Value> manyPointerEquals;
+	List<Value> manyDeepEquals;
+
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/util/benchmark/ValueComparatorBenchmark.java:59]: Could you add a standard main method?

**Pull request merged**